src/components/SignUpHandler.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from "react";
+import axios from "axios";
+import {Modal, ModalContent, ModalHeader, ModalBody, ModalFooter, Button, useDisclosure} from "@nextui-org/react";
+
+const token = process.env.NEXT_PUBLIC_NEXTAUTH_SECRET;
+
+
+const SignUpHandler = ({ email, password, first_name, last_name, confirm_password}) => {
+  const [response, setResponse] = useState(null);
+  const [error, setError] = useState(null);
+  const {isOpen, onOpen, onOpenChange, onClose} = useDisclosure();
+
+
+  const handleSignUp = async () => {
+    try {
+      // console.log("Data being sent:", { first_name,  last_name, email, password, confirm_password }, token); // Log data before sending request
+  
+      const signUpResult = await axios.post(
+        "http://127.0.0.1:8000/api/register",
```

Comment:
fetch in env the Domain, and save to a component to organize the apis

const url = process.env.API_URL;
url+API_REGISTER